src/emmeans.jl

```diff
@@ -0,0 +1,187 @@
+# XXX what about marginalization weights?
+
+_dof(f::Function, model) = f(model)
+_dof(x, _) = x
+
+"""
+    pooled_sem(sems...)
+
+Compute the pooled standard error of the mean.
+
+This corresponds to the square root of the sum of the
+the squared SEMs, which in turn corresponds to the weighted root-mean-square
+of the underlying standard deviations.
+"""
+pooled_sem(sems...) = sqrt(sum(abs2, sems))
+
+# similar to effects (and the underlying math is the same) but
+# the establishment of the reference grid is different
+# we don't allow specifying the "typifier" here -- if you want that,
+# then choose a less full service function
+"""
+    emmeans(model::RegressionModel; eff_col=nothing, err_col=:err,
+            invlink=identity, levels=Dict(), dof=nothing)
+
+Compute estimated martginal means, a.k.a. least-squate (LS) means for a model.
+
+By default, emmeans are computed for each level of each categorical variable
+along with the means of continuous variables. For centered terms, the center
+is used instead of the mean. Alternative levels can be specified with `levels`.
+
+`dof` is used to compute the associated degrees of freedom for a particular
+margin. For regression models, the appropriate degrees of freedom are
+the same degrees of freedom as used for the Wald tests on the coefficients.
+These are typically the residual degrees of freedom (e.g., via `dof_residual`).
+If `dof` is a function, then it is called on `model` and filled elementwise
+into the `dof` column. Alternatively, `dof` can be specified as a single number
+or vector of appropriate length. For example, in a mixed effect model with
+a large number of observations, `dof=Inf` may be appropriate.
+
+`invlink`, `eff_col` and `err_col` work exactly as in [`effects!`](@ref).
+
+Estimated marginal means are closely related to effects and can be viewed as a
+generalization of least-square means. The functionality here is a convenience
```

Comment:
You've said they're the same elsewhere so how is it a generalization?

Reply:
It's possible to define EM means to include things like marginalizing at the end instead of just computing the effect/LS mean. For example, if you had something like `y ~ x * w` and wanted to compute the marginal effect of `x`, you could do:

1. take mean of `w` and compute effects for `x` there
2. compute effects of `x` at extrema of `w` and then compute those estimates into a single, marginal estimate
3. compute effects of `x` at all values of `w` and integrate that out.

For well-sampled models and everything linear, these all work out to be the same thing, but they don't if there is anything nonlinear there. 

See also 
- https://stackoverSflow.com/questions/72201683/how-to-create-a-range-out-of-a-continuous-variable-in-julia
- [SAS LSMESTIMATE](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_glimmix_syntax14.htm)

Reply:
Don't tell me, tell the user :P